nova/objects/console_auth_token.py

```diff
@@ -67,9 +67,19 @@ def access_url(self):
                 # top-level 'token' query parameter was removed. The 'path'
                 # parameter is supported in older noVNC versions, so it is
                 # backward compatible.
-                qparams = {'path': '?token=%s' % self.token}
-                return '%s?%s' % (self.access_url_base,
-                                  urlparse.urlencode(qparams))
+                # NOTE(mariusleu): If the path parameter is already
+                # configured in self.access_url_base, then we should
+                # just append the token to it.
+                url_parts = list(urlparse.urlparse(self.access_url_base))
+                query_params = {}
+                # Transform query_params into simple key-value dictionary
```

Comment:
This is necessary to use urlencode on it? If yes, please comment so. Otherwise, we should not touch more than the `path` parameter.

Reply:
This is necessary, because `parse_qs` returns `dict({key: [value,...]})`, even if the query parameter has a single value.
Then, if you urlencode the same dict, you will get a final url like `"https://example.com?key=[value]"`

Reply:
Sorry, I meant comment in the code :)